Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM python:3 as base
+WORKDIR /app
+COPY ./poetry.toml /app
+COPY ./pyproject.toml /app
+RUN pip install poetry && poetry install
```

Comment:
Minor but it's slightly nicer to install poetry according to [the official docs](https://python-poetry.org/docs/) rather than through pip to remove our dependency on that, though it's not quite as clean to write